### ISSUE TITLE
unnecessary_getters_setters false positive/inconsistent with null-aware assignment operator

### ISSUE THREAD

**Issue by nelsyeung** [dart-lang/linter#2402]
This is giving `unnecessary_getters_setters`, but I believe it shouldn't be:

```dart
bool get foo => _foo; // Avoid wrapping fields in getters and setters just to be "safe".
set foo(bool b) =>  _foo ??= b; // Avoid wrapping fields in getters and setters just to be "safe".
bool _foo;
```

However, this doesn't raise the warning, even though they are effectively the same (i.e., there's an inconsistency):

```dart
bool get foo => _foo;
set foo(bool b) =>  _foo = _foo ?? b;
bool _foo;
```
